app/models/mixins/event_mixin.rb

```diff
@@ -63,7 +63,10 @@ def find_one_event(assoc, order)
 
   module ClassMethods
     def ems_event_filter_column
-      @ems_event_filter_column ||= reflect_on_association(:ems_events).try(:foreign_key) || name.foreign_key
+      @ems_event_filter_column ||=
+        reflect_on_association(:ems_events).try(:foreign_key)    ||
+        reflect_on_association(:event_streams).try(:foreign_key) || # All Events are Ems Events for some classes right now
```

Comment:
Specifically, PhysicalSwitch defines only event_streams.

Reply:
This is not right, I don't think.  EventStream is the base class for all event types, both native provider events and synthetic events; EmsEvent is the sub class for only native events, MiqEvent is the subclass for synthetic events, etc.  There are other subclasses for other types of events that we _don't_ want.

```ruby
EventStream.hierarchy(&:to_s)
=> {"RequestEvent"=>{}, "MiqEvent"=>{}, "CustomEvent"=>{}, "CustomButtonEvent"=>{}, "EmsEvent"=>{}}
```

I think Physical Switch is defined wrong and we need to fix it.

Reply:
I can hold off on fixing this issue for now until I have cycles to add the ems_events association to all the possible timeline classes.  It might just be physical switches but I'll need to find out.